framework/resource/resource.go

```diff
@@ -543,11 +543,19 @@ type ResourceOption func(*Resource)
 
 // WithName sets the resource name. This name is used in output meant to be
 // consumed by a user so it should be descriptive but short, such as
-// "security group".
+// "security group". It must be unique among resources you create.
 func WithName(n string) ResourceOption {
 	return func(r *Resource) { r.name = n }
 }
 
+// WithType sets the type of the resource according to the platform.
+// Example types include container, instance, autoscaling group, pod, etc.
+// Multiple resources may share the same type, and generally one resource
+// will have a type that matches the DeclaredResource corresponding to this resource.
```

Comment:
Docs question. This is optional, presumably? If its not set what happens?

Reply:
Ah good catch! This is optional, and defaults to the resource name. I added this so that plugins will be able to create multiple resources of the same type in the future, but have them all have the same "TYPE" column in the CLI. 

If it's not specified, it defaults to the resource name, because most of the time plugins will only create one resource of each type and use the name _as_ the type, as we are now in all of our plugins.